test/src/main/java/org/sample/NaiveImplementation.java

```diff
@@ -0,0 +1,52 @@
+package org.sample;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.Math;
+
+public class NaiveImplementation {
+  private static double square(double num) {
+    return (num * num);
+  }
+
+  private static double distanceSquared(Coordinates start, Coordinates end) {
+    return square(start.getLng() - end.getLng()) + square(start.getLat() - end.getLat());
+  }
+
+  private static double distanceFromSegment(Coordinates start, Coordinates end, Coordinates point) {
+    double segmentDistanceSquared = distanceSquared(start, end);
+
+    if (segmentDistanceSquared == 0) {
```

Comment:
nit: I would avoid checking equality on double types since this will almost never be true. Try checking against an epsilon value, like segmentDistanceSquared < 1e-8.